### PR TITLE
[Xaml] support arrays as x:Arguments

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

 - Supports x:Array as x:Arguments, for both Xaml and XamlC.
 - Fix a bug with some collections types
 - x:Array is now compiled, so no longer requires reflection call to create the array

### Bugs Fixed ###

- ask @jamesmontemagno 

### Behavioral Changes ###

UISegmentedControls in Xaml Native Views

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs [hunk 1]
```diff
@@ -86,20 +83,25 @@ public static bool InheritsFromOrImplements(this TypeReference typeRef, TypeRefe
 		{
 			var arrayInterfaces = new[]
 			{
-				"System.IEnumerable",
+				"System.Collections.IEnumerable",
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs]: stupid me

File: Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs [hunk 2]
```diff
@@ -86,20 +83,25 @@ public static bool InheritsFromOrImplements(this TypeReference typeRef, TypeRefe
...
 			{
-				"System.IEnumerable`1",
+				"System.Collections.IEnumerable`1",
 				"System.Collections.Generic.IList`1",
 				"System.Collections.Generic.IReadOnlyCollection<T>",
 				"System.Collections.Generic.IReadOnlyList<T>",
 				"System.Collections.Generic.Collection<T>"
 			};
 
+			if (typeRef.IsArray && baseClass.IsArray) {
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs]: if both are arrays, say object[] and string[], resolve the type and apply the test on object and string

File: Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs [hunk 3]
```diff
@@ -109,7 +111,9 @@ public static bool InheritsFromOrImplements(this TypeReference typeRef, TypeRefe
 				    baseClass.IsGenericInstance &&
 				    (baseClass as GenericInstanceType).GenericArguments[0].FullName == arrayType.FullName)
 					return true;
+				return false;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs]: stupid me, again

**Comment by jassmith** [comment]
👍

**Pull request merged**